arch-snapshot/arch.md

```diff
@@ -0,0 +1,2921 @@
+* Instead of humans, machines are the ones browsing offers, negotiating, and
+  sealing deals.
+* Offers are structured rather than free-form text.
+* Golem provides mechanisms to execute deals, such as running workloads on
+  remote machines.
+
+### libp2p for Utilizing Third-Party Resources
+
+Golem Network is comparable to [libp2p](https://libp2p.io), as it consists of
+open-source frameworks and libraries that collectively form a P2P network stack.
+The key differences include:
+
+* Golem focuses on a decentralized marketplace for compute resources rather than
+  being generic.
+* Providers can begin using Golem without writing code, thanks to ready-made
+  binaries for both selling compute power. Requestors can either utilize
+  [declarative
+  DSL](https://docs.golem.network/docs/creators/dapps/creating-golem-dapps) or
+  utulize SDKs for obtaining resources.
+
+## What Golem Is Not
+
+To further clarify what Golem Network is, it may help to define what it is not.
+
+### A Compute Resource Provider
+
+Golem Network does not own the compute resources available on the market.
+Transactions occur directly between buyers and sellers.
+
+### A Middleman or Market Maker
+
+Golem does not mediate or participate in the buying and selling of compute
+resources. Deals are made directly between the involved parties.
+
+### A Blockchain-Based Hosting Service
+
+Just like Golem is not a cloud provider it is also not a higher level services
+provider (e.g. Wordpress or web hosting).
+
+While such services can be built on top of Golem (e.g.,
+[modelserve.ai](https://modelserve.ai/)), the network itself is not
+intended to offer this directly.
+
+## Actors
+
+In Golem Network, there are two primary types of actors. While this is a
+simplification, it is sufficient to grasp the core abstractions introduced by
+the Golem Network.
+
+### Provider
+
+A **Provider** is a seller, someone who rents out their computing resources.
+This could be as simple as renting out a personal laptop, or as extensive as
+offering the resources of an entire data center.
+
+The Provider sets the price for their resources, selects preferred payment
+methods, and configures the necessary software, either provided by or built on
+top of the Golem Network, to run on their machines. Once set up, these resources
+can be utilized by others, and the Provider is compensated for their usage.
+
+### Requestor
+
+A **Requestor** is buyer, someone who wants to borrow computing resources. They
+might do this to experiment with hardware they don’t own, to speed up
+computations that would take too long on a single machine, or even to build a
+service powered by resources from multiple Providers.
+
+The Requestor needs to have a certain level of technical knowledge to interact
+with Golem Network's APIs (e.g., using the Python or Node.js SDKs). Through
+these APIs, they specify what resources they require and how they intend to pay
+for them.
```

Comment:
This is not necessarily true; a Requestor (i.e. the entity demanding services and paying for them) may also have someone else write them the software needed.